kafka/src/main/scala/hydra/kafka/ingestors/KafkaIngestor.scala

```diff
@@ -16,26 +16,46 @@
 
 package hydra.kafka.ingestors
 
-import hydra.core.ingest.Ingestor
 import hydra.core.ingest.RequestParams._
+import hydra.core.ingest.{HydraRequest, Ingestor}
+import hydra.core.monitor.HydraMetrics
 import hydra.core.protocol._
-import hydra.kafka.producer.{ KafkaProducerSupport, KafkaRecordFactories }
+import hydra.kafka.producer.{KafkaProducerSupport, KafkaRecordFactories}
+
+import scala.util.Try
 
 /**
- * Sends JSON messages to a topic in Kafka.  In order for this handler to be activated.
- * a request param "Hydra-kafka-topic" must be present.
- *
- */
+  * Sends JSON messages to a topic in Kafka.  In order for this handler to be activated.
+  * a request param "Hydra-kafka-topic" must be present.
+  *
+  */
 class KafkaIngestor extends Ingestor with KafkaProducerSupport {
+  import KafkaIngestor._
 
   override val recordFactory = new KafkaRecordFactories(schemaRegistryActor)
 
-  //todo: Validate topic Name Topic.validate(topic)
   ingest {
     case Publish(request) =>
       val hasTopic = request.metadataValue(HYDRA_KAFKA_TOPIC_PARAM).isDefined
       sender ! (if (hasTopic) Join else Ignore)
 
     case Ingest(record, ackStrategy) => transport(record, ackStrategy)
   }
+
+  override def validateRequest(request: HydraRequest): Try[HydraRequest] = {
+    val topic = request.metadataValue(HYDRA_KAFKA_TOPIC_PARAM).getOrElse("topicMissing")
+
+    HydraMetrics.incrementGauge(
+      lookupKey = ReconciliationGaugeName + s"_$topic",
+      metricName = ReconciliationMetricName,
+      tags = Seq("ingestType" -> "kafka", "topic" -> topic)
```

Comment:
so we still need the ack strategy on here if we're going to do monitoring, we can really only alert on metrics with a persisted ack level

Reply:
Check out the KafkaTransport class.  By the time we receive the metadata the ack strategy is gone.  A request has an ack strategy while a response doesn't.